ferrilata-devilline/Repositories/LevelRepository.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using ferrilata_devilline.Models.DAOs;
+using Microsoft.EntityFrameworkCore;
+
+namespace ferrilata_devilline.Repositories
+{
+    public class LevelRepository : ILevelRepository
+    {
+        private readonly ApplicationContext _applicationContext;
+
+        public LevelRepository(ApplicationContext applicationContext)
+        {
+            _applicationContext = applicationContext;
+        }
+        public void SaveOrUpdate(Level level)
+        {
+            if (_applicationContext.Levels.Find(level.LevelId) == null)
+            {
+                _applicationContext.Levels.Add(level);
+            }
+
+            _applicationContext.Update(level);
```

Comment:
I think this should go to an else because if it doesn't exist you are adding it and updating it too which is unnecessary.